iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1118,7 +1116,9 @@ void OnTwinResponse(Message possibleResponse)
                      settings.SslProtocols,
                      settings.CertificateRevocationCheck,
                      certs,
-                     iotHubConnectionString.HostName);
+                     clientIdentity.HostName);
+
+                var iotHubConnectionInfo = clientIdentity as IotHubConnectionInfo;
```

Comment:
What if it isn't and this comes back null?

Reply:
That's a good point. I'll update the IClientConfiguration interface to reflect all the properties and methods that any class down the pipeline would expect it to have. We won't need to cast it, then.